includes/class-newspack-blocks.php

```diff
@@ -839,22 +839,28 @@ public static function get_term_classes( $post_id ) {
 		$tags = get_the_terms( $post_id, 'post_tag' );
 		if ( ! empty( $tags ) ) {
 			foreach ( $tags as $tag ) {
-				$classes[] = 'tag-' . $tag->slug;
+				if ( $tag && isset( $tag->slug ) ) {
+					$classes[] = 'tag-' . $tag->slug;
+				}
 			}
 		}
 
 		$categories = get_the_terms( $post_id, 'category' );
 		if ( ! empty( $categories ) ) {
 			foreach ( $categories as $cat ) {
-				$classes[] = 'category-' . $cat->slug;
+				if ( $cat && isset( $cat->slug ) ) {
+					$classes[] = 'category-' . $cat->slug;
+				}
 			}
 		}
 
 		foreach ( self::get_custom_taxonomies() as $tax ) {
 			$terms = get_the_terms( $post_id, $tax['slug'] );
 			if ( ! empty( $terms ) ) {
 				foreach ( $terms as $term ) {
-					$classes[] = $term->taxonomy . '-' . $term->slug;
+					if ( $term && isset( $term->taxonomy ) && isset( $term->slug ) ) {
```

Comment:
```suggestion
					if ( isset( $term->taxonomy, $term->slug ) ) {
```

Not a blocker, just an FYI that this check would suffice without having to check `$term`. Also, just in case it makes sense, using a `! empty` check here might also make sense to not account empty slugs.

Reply:
Cool. I didn't know it's enough 👍